new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DbUtils.java

```diff
@@ -0,0 +1,447 @@
+   * @param dbUrl URL of DB
+   * @throws Exception if any error occurs when setting up RCU database
+   */
+
+  public static void setupDBandRCUschema(String dbImage, String fmwImage, String rcuSchemaPrefix, String dbNamespace,
+      int dbPort, String dbUrl, boolean isUseSecret) throws ApiException {
+
+    // create pull secrets when running in non Kind Kubernetes cluster
+    if (isUseSecret) {
+      CommonTestUtils.createDockerRegistrySecret(OCR_USERNAME, OCR_PASSWORD,
+          OCR_EMAIL, OCR_REGISTRY, OCR_SECRET_NAME, dbNamespace);
+    }
+
+    logger.info("Start Oracle DB with dbImage: {0}, imagePullPolicy: {1}, dbPort: {2}, "
+        + "dbNamespace: {3}", dbImage, dbPort, dbNamespace);
+    startOracleDB(dbImage, dbPort, dbNamespace, isUseSecret);
+    logger.info("Create RCU schema with fmwImage: {0}, rcuSchemaPrefix: {1}, imagePullPolicy: {2}, "
+        + "dbUrl: {3}, dbNamespace: {4}", fmwImage, rcuSchemaPrefix, dbUrl, dbNamespace);
+    createRcuSchema(fmwImage, rcuSchemaPrefix, dbUrl, dbNamespace, isUseSecret);
+
+  }
+
+  /**
+   * Start Oracle DB pod and service in the specified namespace.
+   *
+   * @param dbBaseImageName full image name for DB deployment
+   * @param dbPort NodePort of DB
+   * @param dbNamespace namespace where DB instance is going to start
+   */
+  public static void startOracleDB(String dbBaseImageName, int dbPort, String dbNamespace, boolean isUseSecret)
+      throws ApiException {
+
+    Map labels = new HashMap<String, String>();
+    labels.put("app", "database");
+
+    Map limits = new HashMap<String, String>();
+    limits.put("cpu", "2");
+    limits.put("memory", "10Gi");
+    limits.put("ephemeral-storage", "8Gi");
+    Map requests = new HashMap<String, String>();
+    requests.put("cpu", "500m");
+    requests.put("ephemeral-storage", "8Gi");
+
+    //create V1Service for Oracle DB
+    oracleDBService = new V1Service()
+        .apiVersion("v1")
+        .kind("Service")
+        .metadata(new V1ObjectMeta()
+            .name("oracledb")
+            .namespace(dbNamespace)
+            .labels(labels))
+        .spec(new V1ServiceSpec()
+            .ports(Arrays.asList(
+                new V1ServicePort()
+                    .name("tns")
+                    .port(1521)
+                    .protocol("TCP")
+                    .targetPort(new IntOrString(1521))
+                    .nodePort(dbPort)))
+            .selector(labels)
+            .sessionAffinity("None")
+            .type("LoadBalancer"));
+
+    logger.info("Create service for Oracle DB service in namespace {0}", dbNamespace);
+    boolean serviceCreated = assertDoesNotThrow(() -> Kubernetes.createService(oracleDBService),
+        String.format("Create service failed with ApiException for oracleDBService in namespace %s",
+            dbNamespace));
+    assertTrue(serviceCreated, String.format(
+        "Create service failed with ApiException for oracleDBService in namespace %s ", dbNamespace));
```

Comment:
It comes to here when createService failed without throwing exception.

Reply:
fixed